src/main/java/feign/error/AnnotationErrorDecoder.java

```diff
@@ -109,6 +112,20 @@ public AnnotationErrorDecoder build() {
       return methodErrorHandlerMap;
     }
 
+    Optional<ErrorHandling> readErrorHandlingIncludingInherited(Class<?> apiType) {
+      if (apiType.isAnnotationPresent(ErrorHandling.class)) {
+        return Optional.of(apiType.getAnnotation(ErrorHandling.class));
+      }
+      for (Class<?> parentInterface : apiType.getInterfaces()) {
```

Comment:
This will only support classes that inherit directly.  This will not catch scenarios where a class implementation extends another class.  For example, this will work:

```java
@ErrorHandling
public interface Parent {}

// @ErrorHandling will be inherited here.
public interface Child extends Parent {}
```

But this will not work
```java
@ErrorHandling
public interface Parent {}

// Error Handling will be here
public class Child implements Parent {}

// Error Handling will not be, due to getInterfaces()
public class Grandchild extends Child {}
```

This might be what you want.  In this case, using `getInterfaces` should be sufficient for most, if not all use cases.  Something to keep in mind.

Reply:
thanks for catching this - resolved with added test